src/pages/_app.tsx

```diff
@@ -44,6 +44,28 @@ function App({ Component, pageProps }: AppProps) {
 
    
   }, [router]);
+
+  useEffect(() => {
+    let userInfo;
+
+   
+    if (typeof window !== "undefined" && window.localStorage) {
+      const adminInfo = localStorage.getItem("adminInfo");
+      if (adminInfo && adminInfo !== "undefined") {
+        userInfo = JSON.parse(adminInfo || "{}");
+      }
+
+    }
+    if (userInfo?.role !== Role.ADMIN && userInfo?.role !== Role.CENTRAL_ADMIN && userInfo?.role !== Role.SCTA && userInfo?.role !== Role.CCTA && router.pathname !== "/unauthorized" &&  router.pathname !== "/login" &&  router.pathname !== "/logout") {
```

Comment:
const unauthorizedRoles = [Role.ADMIN, Role.CENTRAL_ADMIN, Role.SCTA, Role.CCTA];
const excludedPaths = ["/unauthorized", "/login", "/logout"];

if (!unauthorizedRoles.includes(userInfo?.role) && !excludedPaths.includes(router.pathname)) {
  router.push({
    pathname: '/unauthorized',
    query: { role: userInfo?.role },
  });
}
@AkshataKatwal16 can u try to optimize this